lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -485,6 +583,14 @@ void FIRRTLModuleLowering::runOnOperation() {
 
           state.oldToNewModuleMap[&op] = loweredMod;
           modulesToProcess.push_back(module);
+          // Lower all the alias types.
+          module.walk([&](Operation *op) {
+            for (auto res : op->getResults()) {
+              if (auto aliasType =
+                      type_dyn_cast<BaseTypeAliasType>(res.getType()))
+                state.lowerType(aliasType, op->getLoc());
+            }
+          });
```

Comment:
Walking entire module could be heavy so it would be nice to paralelly accumulate type aliases (without constructing type decl) in the future.

Reply:
That is a good idea, but as accumulation in parallel will result in non-deterministic order, we need to order them before lowering. Will try it in a separate PR.